operator-pipeline-images/operatorcert/github.py

```diff
@@ -336,3 +343,117 @@ def close_pull_request(
     """
     pull_request.edit(state="closed")
     return pull_request
+
+
+def copy_branch(
+    github_client: Github,
+    src_repo_name: str,
+    src_branch_name: str,
+    dest_repo_name: str,
+    dest_branch_name: str,
+) -> None:
+    """
+    Copy a branch from Source Repository to Destination Repository.
+
+    Args:
+        github_client (Github): A Github API client
+        src_repo_name (str): The source repository name in the format "organization/repository"
+        src_branch_name(str): The name of the branch to copy
+        dest_repo_name(str): The destination repository name in the format "organization/repository"
+        dest_branch_name(str): The name of the destination branch
+    """
+
+    try:
+        src_repository = github_client.get_repo(src_repo_name)
+        dest_repository = github_client.get_repo(dest_repo_name)
+
+        base_dest_branch = dest_repository.get_branch("main")
+        base_dest_commit_sha = base_dest_branch.commit.sha
+
+        ref = f"refs/heads/{dest_branch_name}"
+        dest_repository.create_git_ref(ref=ref, sha=base_dest_commit_sha)
+
+        tree_elements = []
+
+        def collect_files_recursive(path: str = "") -> None:
+            """
+            Recursively fetch and prepare files from the source repo.
+            Args:
+                path (str): Content file path
+            """
+            contents = src_repository.get_contents(path, ref=src_branch_name)
+
+            if isinstance(contents, list):
+                for content in contents:
+                    collect_files_recursive(content.path)
+            else:
+                file_data = contents.decoded_content
+                element = InputGitTreeElement(
+                    path=contents.path,
+                    mode="100644",
+                    type="blob",
+                    content=file_data.decode("utf-8"),
```

Comment:
This won't work with files that are not regular files (for example links) or generic binary files (which cannot be decoded as utf-8 text).
This is not ideal and can cause issues, but I don't think we can fix this without cloning the source repo locally an pushing to the destination; however this might be enough for what we need to achieve in the short term and we may want to fix this down the road.
Can you please note this in the docstring?

Reply:
Updated the doc string with the limitation details.